docs/proposals/compatibility-mode.md

```diff
@@ -0,0 +1,92 @@
+| OCI spec 1.1 without referrers API      | yes               | yes            | 
+| OCI spec 1.1 with referrers API support | yes               | yes            | 
+
+If users want to force pushing a specific version of OCI artifact manifest to a registry, they can use `--image-spec v1.1-artifact`. An OCI artifact manifest will be packed and uploaded. Users might choose it for security requirements, such as pushing a signature to a registry without changing its digest. For example:
+
+```bash
+oras push localhost:5000/hello-artifact:v1 \
+--image-spec v1.1-artifact
+--artifact-type sbom/example \
+sbom.json 
+```
+
+If users want to force pushing an OCI image manifest, no matter whether the registry is compliant with the OCI Spec v1.0 or v1.1, using `--image-spec v1.1-image` will only upload the OCI image manifest to a registry. This option is useful when users have concerns to use OCI artifact manifest or need to migrate content to OCI v1.0 compliant registry. For example:
+
+```bash
+oras push localhost:5000/hello-artifact:v1 \
+--image-spec v1.1-image \
+--artifact-type sbom/example \
+sbom.json \
+```
+
+### Configure compatibility with OCI registry using a flag `--distribution-spec`
+
+Based on the Referrers API status in the registry, users can use flag `--distribution-spec <spec version>-<api>-<option>` to configure compatibility with registry. 
+
+| registry support                        |  v1.1-referrers-api | v1.1-referrers-tag |
+| :-------------------------------------- | --- | --- | 
+| OCI spec 1.0                            | no  | yes[^footnote] |
+| OCI spec 1.1 without referrers API      | no  | yes |
+| OCI spec 1.1 with referrers API support | yes | yes |
+
+> [^footnote]: It only works when the registry returns error code 404 for [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#listing-referrers). 
+
+Using a flag `--distribution-spec v1.1-referrers-api` to disable backward compatibility. It only allows uploading OCI artifact manifest to OCI v1.1 compliant registry with Referrers API enabled. This is the most strict option for setting compatibility with the registry. Users might choose it for security requirements. 
+
+For example, using this flag, ORAS will attach OCI artifact manifest only to an OCI v1.1 compliant registry with Referrers API enabled and no further actions for maintaining artifact references in registries.  
+
+```bash
+oras attach localhost:5000/hello-artifact:v1 \
+--artifact-type sbom/example \
+--distribution-spec v1.1-referrers-api \
+sbom.json 
+```
+
+Using `--distribution-spec v1.1-referrers-tag` to enable maximum backward compatibility with the registry. It will upload OCI image manifest and [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#referrers-tag-schema) regardless of whether the registry complies with the OCI Spec v1.0 or v1.1 or support Referrers API. For example: 
```

Comment:
To enable maximum backward compatibility with the registry, both flags are required.